src/NLog/MappedDiagnosticsLogicalContext.cs

```diff
@@ -140,12 +151,23 @@ public static object GetObject(string item)
         /// <param name="item">Item name.</param>
         /// <param name="value">Item value.</param>
         /// <returns>>An <see cref="IDisposable"/> that can be used to remove the item from the current logical context.</returns>
-        public static IDisposable SetScoped(string item, string value)
+        public static IDisposable SetScoped<T>(string item, T value)
```

Comment:
please move the generic below the non-generic, less potential conflict then

Reply:
Fixed